### PR TITLE
Remove errorCodeIfNone annotation for normalizeError

### LINKED ISSUE #8102: Remove errorCodeIfNone capability from normalizeError
`normalizeError` is a function which takes an `unknown` error as input and returns a valid Fluid Error, optionally annotating it with `IFluidErrorAnnotations`.  One annotation type is `errorCodeIfNone`, which will stamp the given `fluidErrorCode` on the resulting error if it wasn't already a valid Fluid Error.

Looking back at my notes from the design process that lead to this point, there's not necessarily a clear purpose this serves, other concepts morphed into it in the end.  And revisiting the idea, I expect it to lead to ambiguity and confusion when trying to interpret the logs.  Perhaps the point was to try to get everything with an error code? But it's actually kind of nice to have errors that we didn't raise ourselves left with no error code.  Gives a much cleaner semantic around what `fluidErrorCode` means.

### PR DESCRIPTION
**Pull request by markfields**
Fixes #8102, see that issue for context on why.  Fortunately it was not hardly used at all.

The only change we'll see to our data is that DataProcessingErrors will no longer have an error code set. The error code like thing passed to `CreateProcessingError` will go in a new prop `dataProcessingCodepath` and will be present regardless of whether the provided error was already a valid Fluid Error.  This way, we'll be able to use the presence or absence of `fluidErrorCode` to distinguish errors we originated v. that came to us from elsewhere (related to the discussion on `errorSource` property)

### CODE REVIEW

File: packages/utils/telemetry-utils/src/errorLogging.ts [hunk 1]
```diff
@@ -117,11 +115,10 @@ class SimpleFluidError implements IFluidErrorBase {
 /** For backwards compatibility with pre-fluidErrorCode valid errors */
 function patchWithErrorCode(
     legacyError: Omit<IFluidErrorBase, "fluidErrorCode">,
-    errorCode: string = "<error predates fluidErrorCode>",
 ): asserts legacyError is IFluidErrorBase {
-    const patchMe: { fluidErrorCode?: string } = legacyError as any;
+    const patchMe: { -readonly [P in "fluidErrorCode"]?: IFluidErrorBase[P] } = legacyError as any;
     if (patchMe.fluidErrorCode === undefined) {
-        patchMe.fluidErrorCode = errorCode;
+        patchMe.fluidErrorCode = "<error predates fluidErrorCode>";
```
Comment (anthony-murphy) [inline packages/utils/telemetry-utils/src/errorLogging.ts:121]: this the wrong format for fluid error code. shouldn't we just not add the property

Comment (anthony-murphy) [reply packages/utils/telemetry-utils/src/errorLogging.ts:121]: in my mind the existence of the property or not may be important. so we should never add it beyond initial creation

File: packages/utils/telemetry-utils/src/errorLogging.ts [hunk 2]
```diff
@@ -150,7 +147,7 @@ export function normalizeError(
     const { message, stack } = extractLogSafeErrorProperties(error, false /* sanitizeStack */);
     const fluidError: IFluidErrorBase = new SimpleFluidError({
         errorType: "genericError", // Match Container/Driver generic error type
-        fluidErrorCode: annotations.errorCodeIfNone ?? "none",
+        fluidErrorCode: "none",
```
Comment (anthony-murphy) [inline packages/utils/telemetry-utils/src/errorLogging.ts:150]: again. lets not add it

File: packages/loader/container-loader/src/container.ts [hunk 3]
```diff
@@ -869,7 +869,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             assert(!canRetryOnError(error), 0x24f /* "retriable error thrown from attach()" */);
 
             // add resolved URL on error object so that host has the ability to find this document and delete it
-            const newError = DataProcessingError.wrapIfUnrecognized(
+            const newError = CreateProcessingError(
                 error, "errorWhileUploadingBlobsWhileAttaching", undefined);
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/container.ts:873]: this whole flow in weird. why this change at all

File: packages/loader/container-utils/src/error.ts [hunk 4]
```diff
@@ -113,27 +113,30 @@ export class DataProcessingError extends LoggingError implements IErrorBase, IFl
...
     static wrapIfUnrecognized(
         originalError: any,
-        errorCodeIfNone: string,
+        dataProcessingCodepath: string,
         message?: ISequencedDocumentMessage,
     ): IFluidErrorBase {
         const newErrorFn = (errMsg: string) => {
-            const dpe = new DataProcessingError(errMsg, errorCodeIfNone);
-            dpe.addTelemetryProperties({ untrustedOrigin: 1}); // To match normalizeError
+            const dpe = new DataProcessingError(errMsg, "none" /* fluidErrorCode */);
```
Comment (anthony-murphy) [inline packages/loader/container-utils/src/error.ts:125]: i think errorCodeIfNone still applies in this case. the caller is giving a value, but you are setting it to none. seem strange

File: packages/loader/container-utils/src/error.ts [hunk 5]
```diff
@@ -113,27 +113,30 @@ export class DataProcessingError extends LoggingError implements IErrorBase, IFl
...
+            dpe.addTelemetryProperties({ untrustedOrigin: 1}); // To match normalizeError. Redundant with "none" above
             return dpe;
         };
 
         // Don't coerce if already has an errorType, to distinguish unknown errors from
         // errors that we raised which we already can interpret apart from this classification
         const error = isValidLegacyError(originalError) // also accepts valid Fluid Error
-            ? normalizeError(originalError, { errorCodeIfNone })
```
Comment (anthony-murphy) [inline packages/loader/container-utils/src/error.ts:133]: i agree not passing it here

File: packages/loader/container-utils/src/error.ts (hunk 4 above)
Comment (anthony-murphy) [reply packages/loader/container-utils/src/error.ts:125]: maybe i get it. i will still prefer undefined over none

File: packages/loader/container-loader/src/container.ts (hunk 3 above)
Comment (markfields) [reply packages/loader/container-loader/src/container.ts:873]: I followed up with Wes here yesterday, he switched it to `normalizeError`. Didn't look like a data processing error to me.

File: packages/utils/telemetry-utils/src/errorLogging.ts (hunk 1 above)
Comment (markfields) [reply packages/utils/telemetry-utils/src/errorLogging.ts:121]: WF -- This is a temporary measure to backfill the presence of the property in cross-version scenarios.  I'm not going to change it now, it will be phased out in a few more months anyway and is useful to be able to search for this string to watch for when we can do so.

File: packages/utils/telemetry-utils/src/errorLogging.ts (hunk 2 above)
Comment (markfields) [reply packages/utils/telemetry-utils/src/errorLogging.ts:150]: Sure. This means we can't distinguish which errors have been "touched" via `normalize`, but I can't think of why we would care, the only thing that means now is that there may be other telemetry props added.

I'm going to do `""` to keep the field required for the sake of the type guard, since we can have old/new versions of this code and the type guard in play together.  If we want to make it actually `undefined` I'll stage that in subsequent changes.

Comment (markfields) [reply packages/utils/telemetry-utils/src/errorLogging.ts:150]: Actually the sneaky `untrustedOrigin` is still around to say an error went through `normalize`, not that I think we'll need it 🤷

Comment (anthony-murphy) [reply packages/utils/telemetry-utils/src/errorLogging.ts]: seems fine either way. you can also punt on it for now. if it becomes fluidCallSite, then allowing it to not exist makes sense, and only error we spawn should have it

**Review by anthony-murphy — APPROVED** [review]
